src/no/runsafe/UserControl/database/PlayerDatabase.java

```diff
@@ -81,56 +82,56 @@ public void logPlayerInfo(IPlayer player)
 		database.update(
 			"INSERT INTO player_db (`uuid`,`name`,`joined`,`login`,`ip`) VALUES (?,?,NOW(),NOW(),INET_ATON(?))" +
 				"ON DUPLICATE KEY UPDATE `uuid`=VALUES(`uuid`), `name`=VALUES(`name`), `login`=VALUES(`login`), `ip`=VALUES(`ip`)",
-			((RunsafePlayer)player).getBasePlayer().getUniqueId().toString(), player.getName(), player.getIP()
+			player.getUniqueId().toString(), player.getName(), player.getIP()
 		);
-		dataCache.Invalidate(player.getName());
+		dataCache.Invalidate(player);
 		lookupCache.Purge();
 	}
 
 	public void logPlayerBan(IPlayer player, IPlayer banner, String reason)
 	{
 		database.update(
-			"UPDATE player_db SET `banned`=NOW(), ban_reason=?, ban_by=? WHERE `name`=?",
-			reason, banner == null ? "console" : banner.getName(), player.getName()
+			"UPDATE player_db SET `banned`=NOW(), ban_reason=?, ban_by=? WHERE `uuid`=?",
+			reason, banner == null ? "console" : banner.getName(), player.getUniqueId()
 		);
-		dataCache.Invalidate(player.getName());
+		dataCache.Invalidate(player);
 	}
 
 	public void setPlayerTemporaryBan(IPlayer player, DateTime temporary)
 	{
 		database.update("UPDATE player_db SET temp_ban=? WHERE `name`=?", temporary, player.getName());
-		dataCache.Invalidate(player.getName());
+		dataCache.Invalidate(player);
 	}
 
 	public void logPlayerUnban(IPlayer player)
 	{
 		database.update(
-			"UPDATE player_db SET `banned`=NULL, ban_reason=NULL, ban_by=NULL, temp_ban=NULL WHERE `name`=?",
-			player.getName()
+			"UPDATE player_db SET `banned`=NULL, ban_reason=NULL, ban_by=NULL, temp_ban=NULL WHERE `uuid`=?",
+			player.getUniqueId().toString()

```

Comment:
I tested it and it doesn't implicitly call the toString method, instead it appears to output the import file UUID is in followed by garbled data along with the first 2 lines of the code in the toString method. This specifically:
▒▒ sr java.util.UUID▒▒▒m▒/ J
leastSigBitsJ
              mostSigBitsxp▒
▒D8B▒b▒▒▒fmA▒